packages/medusa/src/api/admin/product-options/[id]/route.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  deleteProductOptionsWorkflow,
+  updateProductOptionsWorkflow,
+} from "@medusajs/core-flows"
+import {
+  AuthenticatedMedusaRequest,
+  MedusaResponse,
+  refetchEntity,
+} from "@medusajs/framework/http"
+
+import {
+  AdminGetProductOptionParamsType,
+  AdminUpdateProductOptionType,
+} from "../validators"
+import { HttpTypes } from "@medusajs/framework/types"
+import { MedusaError } from "@medusajs/framework/utils"
+
+export const GET = async (
+  req: AuthenticatedMedusaRequest<AdminGetProductOptionParamsType>,
+  res: MedusaResponse<HttpTypes.AdminProductOptionResponse>
+) => {
+  const product_option = await refetchEntity({
+    entity: "product_option",
+    idOrFilter: req.params.id,
+    scope: req.scope,
+    fields: req.queryConfig.fields,
+  })
+
+  res.status(200).json({ product_option })
+}
+
+export const POST = async (
+  req: AuthenticatedMedusaRequest<AdminUpdateProductOptionType>,
+  res: MedusaResponse<HttpTypes.AdminProductOptionResponse>
+) => {
+  const existingProductOption = await refetchEntity({
+    entity: "product_option",
+    idOrFilter: req.params.id,
+    scope: req.scope,
+    fields: ["id"],
+  })
+
+  if (!existingProductOption) {
+    throw new MedusaError(
+      MedusaError.Types.NOT_FOUND,
+      `Product option with id "${req.params.id}" not found`
+    )
+  }
```

Comment:
Q: since the workflow will also fetch data, would it make sense to include that either as part of the workflow, or being able to provide prefetched data to the workflow? I am wondering, I would prefer the first one in some ways but let you decide if even needed right now :)